appliance/mariadb/process_test.go

```diff
@@ -0,0 +1,484 @@
+package mariadb
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	. "github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-check"
+	_ "github.com/flynn/flynn/Godeps/_workspace/src/github.com/go-sql-driver/mysql"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/pkg/attempt"
+	"github.com/flynn/flynn/pkg/sirenia/state"
+)
+
+// Hook gocheck up to the "go test" runner
+func Test(t *testing.T) { TestingT(t) }
+
+type MariaDBSuite struct{}
+
+var _ = Suite(&MariaDBSuite{})
+
+func (MariaDBSuite) TestSingletonPrimary(c *C) {
+	p := NewProcess()
+	p.ID = "node1"
+	p.Singleton = true
+	p.Password = "password"
+	p.DataDir = c.MkDir()
+	p.Port = "54320"
+	p.ServerID = 1
+	p.OpTimeout = 30 * time.Second
+	err := p.Reconfigure(&state.Config{Role: state.RolePrimary})
+	c.Assert(err, IsNil)
+
+	err = p.Start()
+	c.Assert(err, IsNil)
+	defer p.Stop()
+
+	conn := connect(c, 0, "")
+	_, err = conn.Exec("CREATE DATABASE test")
+	conn.Close()
+	c.Assert(err, IsNil)
+
+	err = p.Stop()
+	c.Assert(err, IsNil)
+
+	// ensure that we can start a new instance from the same directory
+	p = NewProcess()
+	p.ID = "node1"
+	p.Singleton = true
+	p.Password = "password"
+	p.DataDir = c.MkDir()
+	p.Port = "54320"
+	p.ServerID = 1
+	p.OpTimeout = 30 * time.Second
+	err = p.Reconfigure(&state.Config{Role: state.RolePrimary})
+	c.Assert(err, IsNil)
+	c.Assert(p.Start(), IsNil)
+	defer p.Stop()
+
+	conn = connect(c, 0, "")
+	_, err = conn.Exec("CREATE DATABASE foo")
+	conn.Close()
+	c.Assert(err, IsNil)
+
+	err = p.Stop()
+	c.Assert(err, IsNil)
+}
+
+func instance(n int) *discoverd.Instance {
+	id := fmt.Sprintf("node%d", n)
+	return &discoverd.Instance{
+		ID:   id,
+		Addr: fmt.Sprintf("127.0.0.1:5432%d", n),
+		Meta: map[string]string{
+			"MYSQL_ID":  id,
+			"HTTP_ADDR": fmt.Sprintf("127.0.0.1:%d", 54420+n),
+		},
+	}
+}
+
+func connect(c *C, n int, database string) *sql.DB {
+	dsn := DSN{
+		Host:     fmt.Sprintf("127.0.0.1:%d", 54320+uint16(n)),
```

Comment:
I recently refactored the corresponding postgres tests to not reuse port numbers, as it is unreliable: https://github.com/flynn/flynn/commit/3ba9110c30b771a72a86f32bbea26c00c1e8c21c